packages/stargate/src/signingstargateclient.ts

```diff
@@ -0,0 +1,121 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { fromBase64 } from "@cosmjs/encoding";
+import {
+  buildFeeTable,
+  Coin,
+  CosmosFeeTable,
+  DirectSignSpec,
+  encodeSecp256k1Pubkey,
+  GasLimits,
+  GasPrice,
+  OfflineSigner,
+  StdFee,
+} from "@cosmjs/launchpad";
+import { Int53 } from "@cosmjs/math";
+import { EncodeObject, encodePubkey, makeAuthInfo, makeDirectSignJob, Registry } from "@cosmjs/proto-signing";
+import { Client as TendermintClient } from "@cosmjs/tendermint-rpc";
+
+import { cosmos } from "./codec";
+import { BroadcastTxResponse, StargateClient } from "./stargateclient";
+
+const { TxRaw } = cosmos.tx.v1beta1;
+
+const defaultGasPrice = GasPrice.fromString("0.025ucosm");
+const defaultGasLimits: GasLimits<CosmosFeeTable> = { send: 80000 };
+
+/** Use for testing only */
+export interface PrivateSigningStargateClient {
```

Comment:
Ah, now I see... you use the casts above to expose private variables.

If any of these private variables are security critical, we should do something to make them really private (like wrap in some closure). This may be the case for some signer implementation.

Reply:
Yes. So far I don't believe this applies to our signers - we even expose `mnemonic` as a public property somewhere.